File: lib_internet/src/main/java/world/share/lib_internet/rxjava/BaseDisposableObserver.java
package world.share.lib_internet.rxjava;


import io.reactivex.observers.DisposableObserver;
import world.share.lib_internet.http.exception.ApiException;
import world.share.lib_internet.http.exception.FactoryException;
import world.share.lib_internet.http.exception.NetErrorBean;


/**
 * 自定义DisposableObserver
 *
 * @author mac
 */
public abstract class BaseDisposableObserver<T> extends DisposableObserver<T> {

    @Override
    public final void onError(Throwable e) {
        ApiException responseThrowable = FactoryException.analysisExcetpion(e);
        NetErrorBean errorBean = new NetErrorBean(responseThrowable.getCode(), responseThrowable.getMessage());
        onError(errorBean);
    }

    public void onError(NetErrorBean errorBean) {
    }

    @Override
    public void onComplete() {
    }
}

File: lib_base/src/main/java/world/share/lib_base/internet/rxjava/BaseDisposableObserver.kt
package world.share.lib_base.internet.rxjava

import world.share.lib_base.internet.http.exception.ApiException
import io.reactivex.observers.DisposableObserver
import world.share.lib_base.internet.http.exception.FactoryException
import world.share.lib_base.internet.http.exception.NetErrorBean

/**
 * 自定义DisposableObserver
 *
 * @author mac
 */
abstract class BaseDisposableObserver<T> constructor() : DisposableObserver<T>() {
    public override fun onError(e: Throwable) {
        val responseThrowable: ApiException? = FactoryException.analysisExcetpion(e)
        val errorBean: NetErrorBean? =
            responseThrowable?.let { NetErrorBean(it.code, responseThrowable!!.message) }
        onError(errorBean)
    }

    fun onError(errorBean: NetErrorBean?) {}
    public override fun onComplete() {}
}